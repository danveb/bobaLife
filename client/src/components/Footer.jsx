import { FaGlobeAmericas, FaWhatsapp, FaEnvelopeOpen, FaRegCreditCard, FaFacebookF, FaInstagram, FaTwitter, FaGithub } from "react-icons/fa"; 
import "./Footer.scss"; 

const Footer = () => {
    return (
        <div id="contact" className="footer-container">
            <div className="footer-left">
            <h3 className="footer-title">Contact Us</h3>
                <div className="footer-contact">
                    <FaGlobeAmericas style={{ marginRight: "10px" }}/>310 N Rodeo Dr, Beverly Hills, CA 90210</div>
                <div className="footer-contact">
                    <FaWhatsapp style={{ marginRight: "10px" }}/>+1 (310) 156-3333
                </div>
                <div className="footer-contact">
                    <FaEnvelopeOpen style={{ marginRight: "10px" }}/>contact@bobalife.com
                </div>
                <div className="footer-contact">
                    <FaRegCreditCard style={{ marginRight: "10px" }}/>Contactless payments accepted
                </div>
            </div>
            <div className="footer-center">
                <h3 className="footer-title">Our Hours</h3>
                <ul className="footer-list">
                    <li className="footer-listItem">Every Day @ 10:00 - 21:00</li>
                </ul>
            </div>
            <div className="footer-right">
                <h3 className="footer-brand">bobaLife</h3>
                <p className="footer-subtitle">Proud to serve a world-class boba carefully crafted by our baristas.</p>
                <div className="footer-socialContainer">
                    <div className="footer-socialIcon" color="3b5999">
                        <a href="https://www.facebook.com" target="_blank" rel="noreferrer"><FaFacebookF /></a>
                    </div>
                    <div className="footer-socialIcon" color="e4405f">
                        <a href="https://www.instagram.com" target="_blank" rel="noreferrer"><FaInstagram /></a>
                    </div>
                    <div className="footer-socialIcon" color="55acee">
                        <a href="https://www.twitter.com" target="_blank" rel="noreferrer"><FaTwitter /></a>
                    </div>
                    <div className="footer-socialIcon" color="000000">
                        <a href="https://www.github.com/danveb" target="_blank" rel="noreferrer"><FaGithub /></a>
                    </div>
                </div>
            </div>    
        </div>
    )
}

export default Footer 