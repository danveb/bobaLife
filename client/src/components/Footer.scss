.footer-container {
    display: flex; 

    @media screen and (max-width: 768px) {
        flex-direction: column; 
    }
}

.footer-left {
    flex: 1; 
    display: flex; 
    flex-direction: column; 
    padding: 30px; 

    @media screen and (max-width: 768px) {
        padding: 20px 20px 10px 20px; 
    }
}

.footer-title {
    margin-bottom: 30px; 

    @media screen and (max-width: 768px) {
        margin-bottom: 20px; 
    }
}

.footer-contact {
    margin-bottom: 20px; 
    display: flex; 
    align-items: center; 

    @media screen and (max-width: 768px) {
        margin-bottom: 10px; 
    }
}

.footer-center {
    flex: 1; 
    padding: 30px; 

    @media screen and (max-width: 768px) {
        padding: 10px 20px; 
    }
}

.footer-list {
    margin: 0; 
    padding: 0; 
    list-style: none; 
    display: flex; 
    flex-wrap: wrap; 
}

.footer-listItem {
    width: 100%; 
    margin-bottom: 10px; 
}

.footer-right {
    flex: 1;
    padding: 30px; 

    @media screen and (max-width: 768px) {
        padding: 10px 20px 30px 20px; 
        
    }
}

.footer-brand {
    margin-bottom: 30px; 

    @media screen and (max-width: 768px) {
        margin-bottom: 20px; 
    }
}

.footer-subtitle {
    margin-bottom: 20px; 
}

.footer-socialContainer {
    display: flex; 
}

.footer-socialIcon {
    width: 40px; 
    height: 40px; 
    border-radius: 50%; 
    display: flex; 
    justify-content: center; 
    align-items: center; 
    flex-direction: column; 
    margin-right: 20px; 

    a {
        color: #fff; 
        display: flex; 
        justify-content: center; 
        align-items: center; 
    }

    &:nth-child(1) {
        background-color: #3b5999;
    }

    &:nth-child(2) {
        background-color: #e4405f; 
    }

    &:nth-child(3) {
        background-color: #55acee;
    }

    &:nth-child(4) {
        background-color: #000000; 
    }
}