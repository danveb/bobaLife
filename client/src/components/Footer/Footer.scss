.footer {
    display: flex; 
    padding: 0 24px 24px 24px; 
    
    .footer-wrapper {
        display: flex;
        width: 100%; 
        background-color: var(--languid-lavender); 

        @media screen and(max-width: 768px) {
            flex-direction: column; 
        }

        @media screen and(max-width: 480px) {
            padding: 10px; 
        }

        .footer-left {
            flex: 1; 
            display: flex; 
            flex-direction: column; 
            padding: 30px; 

            @media screen and (max-width: 768px) {
                padding: 20px 20px 10px 20px; 
            }

            .footer-title {
                margin-bottom: 30px; 

                h3 {
                    font-size: 22px; 
                    font-weight: 300; 
                }
    
                @media screen and (max-width: 768px) {
                    margin-bottom: 20px; 
                }
            }
    
            .footer-contact {
                margin-bottom: 20px; 
                display: flex; 
                align-items: center; 
    
                @media screen and (max-width: 768px) {
                    margin-bottom: 10px; 
                }
            }
        }

        .footer-center {
            flex: 1; 
            padding: 30px; 

            @media screen and (max-width: 768px) {
                padding: 10px 20px; 
            }

            .footer-title {
                margin-bottom: 30px; 

                h3 {
                    font-size: 22px; 
                    font-weight: 300; 
                }
    
                @media screen and (max-width: 768px) {
                    margin-bottom: 20px; 
                }
            }

            .footer-list {
                margin: 0; 
                padding: 0; 
                list-style: none; 
                display: flex; 
                flex-wrap: wrap; 
            }

            .footer-listItem {
                width: 100%; 
                margin-bottom: 10px; 
            }
        }

        .footer-right {
            flex: 1;
            padding: 30px; 

            @media screen and (max-width: 768px) {
                padding: 10px 20px 30px 20px;                 
            }

            .footer-brand {
                margin-bottom: 30px; 

                h3 {
                    font-size: 22px; 
                    font-weight: 300; 
                }
    
                @media screen and (max-width: 768px) {
                    margin-bottom: 20px; 
                }
            }
    
            .footer-subtitle {
                margin-bottom: 20px; 
            }
    
            .footer-socialContainer {
                display: flex; 
                
                .footer-social {
                    width: 40px; 
                    height: 40px; 
                    border-radius: 50%; 
                    display: flex; 
                    justify-content: center; 
                    align-items: center; 
                    flex-direction: column; 
                    margin-right: 20px; 
        
                    a {
                        color: var(--alice-blue); 
                        display: flex; 
                        justify-content: center; 
                        align-items: center; 
    
                        svg {
                            width: 21px; 
                            height: 21px; 
                        }
                    }
        
                    // Potentially add more logos
                    &:nth-child(1) {
                        background-color: var(--rich-black); 
                    }
                }
            }
    
        }
    }
}