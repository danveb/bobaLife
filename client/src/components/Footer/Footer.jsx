import { footerContact, footerHours } from "../../constants/footer"; 
import { GitHub } from "@mui/icons-material"; 
import { FooterContainer, Wrapper, Left, H3, Contact, Center, Right, P, H5, Social, SocialA } from "../../styles/Footer.style";

const Footer = () => {
    return (
        <FooterContainer>
            <Wrapper>
                <Left>
                    <H3>Contact Us</H3>
                    {footerContact.map((item) => (
                        <Contact 
                            key={item.id}
                        >
                            {item.logo}
                            {item.text}
                        </Contact>
                    ))}
                </Left>
                <Center>
                    <H3>Our Hours</H3>
                    {footerHours.map((item) => (
                        <Contact
                            key={item.id}
                        >
                            {item.logo} 
                            {item.text}
                        </Contact>
                    ))}
                </Center>
                <Right>
                    <H3>bobaLife</H3>
                    <P>Proud to serve a world-class boba carefully crafted by our baristas.</P>
                    <H5>© All rights reserved – Danny Bae</H5>
                    <Social>
                        <SocialA href="https://www.github.com/danveb" target="_blank" rel="noreferrer"><GitHub /></SocialA>
                    </Social>
                </Right>
            </Wrapper>
        </FooterContainer>
    )
}

export default Footer 