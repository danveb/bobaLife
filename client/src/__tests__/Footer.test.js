import { render, screen } from "@testing-library/react"; 
import Footer from "../components/Footer/Footer"; 

describe("Footer component", () => {
    // smoke test
    test("renders without crashing", () => {
        render(<Footer />); 
    }); 

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(<Footer />); 
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByText 
    test("should display the P tag", () => {
        render(<Footer />); 
        const paragraph = screen.getByText("All rights reserved", { exact: false });
        expect(paragraph).toBeInTheDocument(); 
    }); 

    // getByRole 
    test("should display anchor tag for GitHub logo", () => {
        render(<Footer />); 
        const anchor = screen.getByRole("link"); 
        expect(anchor).toHaveAttribute("href"); 
    });
});