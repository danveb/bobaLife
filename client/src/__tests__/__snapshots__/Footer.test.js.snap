// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ cHmkLw"
  >
    <div
      class="sc-gsnTZi fExhLj"
    >
      <div
        class="sc-dkzDqf kPVtdG"
      >
        <h3
          class="sc-hKMtZM gtgbeu"
        >
          Contact Us
        </h3>
        <div
          class="sc-eCYdqJ bPZCLf"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LanguageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
          1502 N Milwaukee Ave, Chicago, IL 60622
        </div>
        <div
          class="sc-eCYdqJ bPZCLf"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="WhatsAppIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
            />
          </svg>
          +1 (312) 846-9730
        </div>
        <div
          class="sc-eCYdqJ bPZCLf"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SendIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
            />
          </svg>
          contact@bobalife.com
        </div>
        <div
          class="sc-eCYdqJ bPZCLf"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CreditCardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
            />
          </svg>
          Contactless payments accepted
        </div>
      </div>
      <div
        class="sc-jSMfEi kbINUw"
      >
        <h3
          class="sc-hKMtZM gtgbeu"
        >
          Our Hours
        </h3>
        <div
          class="sc-eCYdqJ bPZCLf"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="QueryBuilderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
          Every Day @ 11:00 - 21:00
        </div>
      </div>
      <div
        class="sc-gKXOVf cPfwEZ"
      >
        <h3
          class="sc-hKMtZM gtgbeu"
        >
          bobaLife
        </h3>
        <p
          class="sc-iBkjds gvMqcN"
        >
          Proud to serve a world-class boba carefully crafted by our baristas.
        </p>
        <h5
          class="sc-ftvSup GVnGs"
        >
          © All rights reserved – Danny Bae
        </h5>
        <div
          class="sc-papXJ bKmLNh"
        >
          <a
            class="sc-jqUVSM fOlkIy"
            href="https://www.github.com/danveb"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
